src/routes/(infoPages)/land/fire/safety/fire-danger-map/+page.server.ts

```diff
@@ -1,47 +1,69 @@
+import { error } from "@sveltejs/kit";
 import parishTopoJSON from "./parishes.json";
 
-import type { FireWeatherMapQuery } from "./$queries.generated";
+import type { FireWeatherMapPageQuery } from "./$queries.generated";
 
 const query = gql`
-  query FireWeatherMap {
-    fireDangerMapCollection(limit: 1) {
+  query FireWeatherMapPage($metadataID: String!, $preview: Boolean = false) {
+    fireDangerMapCollection(
+      where: { pageMetadata: { sys: { id: $metadataID } } }
+      limit: 1
+      preview: $preview
+    ) {
       items {
+        title
         description {
           json
         }
+        pageMetadata {
+          sys {
+            id
+          }
+        }
       }
     }
   }
 `;
 
-export const load = async ({ locals: { getKVClient }, locals: { contentfulClient } }) => {
-  try {
-    const kvClient = await getKVClient();
-    const fireWeatherData = await kvClient.getFireWeatherData();
-
-    if (fireWeatherData === undefined) {
-      throw new Error("Error fetching fire weather data.");
-    }
+export const load = async ({ parent, locals: { getKVClient, contentfulClient } }) => {
+  fetchData: {
+    if (!contentfulClient) break fetchData;
+    const { pageMetadataMap, pathsToIDs } = await parent();
+    const metadataID = pathsToIDs.get("/land/fire/safety/fire-danger-map");
+    if (!metadataID) break fetchData;
+    const pageMetadata = pageMetadataMap.get(metadataID);
+    if (!pageMetadata) break fetchData;
 
-    const pageData = await contentfulClient?.fetch<FireWeatherMapQuery>(printQuery(query));
+    const pageData = await contentfulClient.fetch<FireWeatherMapPageQuery>(printQuery(query), {
+      variables: { metadataID },
+    });
+    if (!pageData?.fireDangerMapCollection?.items) break fetchData;
+    const [fireDangerMapPageData] = pageData.fireDangerMapCollection.items;
 
-    const mapDescription = pageData?.fireDangerMapCollection?.items[0]?.description;
+    try {
+      // Fetch fire weather data from database.
+      const kvClient = await getKVClient();
+      const fireWeatherData = await kvClient.getFireWeatherData();
+      if (fireWeatherData === undefined) {
+        throw new Error("Error fetching fire weather data.");
+      }
 
-    return {
-      fireWeatherData: {
-        ...fireWeatherData,
-        topoJSON: parishTopoJSON,
-      },
-      pageData: {
-        description: mapDescription,
-      },
-    };
-  } catch (error) {
-    const errorMessage = error instanceof Error ? error.message : "An unknown error occurred";
-    return {
-      error: errorMessage,
-    };
+      return {
+        pageMetadata,
+        fireWeatherData: {
+          ...fireWeatherData,
+          topoJSON: parishTopoJSON,
+        },
+        pageData: fireDangerMapPageData,
+      };
+    } catch (error) {
+      const errorMessage = error instanceof Error ? error.message : "An unknown error occurred";
+      return {
+        error: errorMessage,
+      };
+    }
```

Comment:
This block is effectively unchanged, just nested in `fetchData` with the content fetching moved further up.